### PR TITLE
Add support for <group> layer tags

### PR DESCRIPTION
**Pull request by Grant1219**
This is a start for supporting the `<group>` tag added in version 1.0 of the TMX format. Let me know if my approach here doesn't make sense, or if I'm missing something major.

I ran the parse test and it seems to still be working, but I would like to update it to include nested layers within layer groups to make sure everything is functioning.

### CODE REVIEW

**Comment by fallahn** [comment]
It looks sane to me - from what I understand a layer group is not dissimilar to an object group, so a similar approach to parsing feels correct. The parse test doesn't specifically test for layer groups, so it will need updating. If you're prepared to do it that would be fantastic :D I've noticed the CI for msvc has failed, but I'll fix that when I update the Visual Studio solution. Thanks for your efforts!

**Comment by Grant1219** [comment]
@fallahn Oh is the MSVC build failing for a reason unrelated to my code? I made a couple fixups, but it didn't seem to resolve the build errors. Unfortunately I don't have access to MSVC to dig into it.

Anyway, I'll probably get around to updating the tests tomorrow with layer groups. Thanks for being so responsive. :)

**Comment by fallahn** [comment]
No problem! Visual Studio is my main environment so I can hop into it as soon as I have a few moments.

File: tmxlite/include/tmxlite/LayerGroup.hpp [hunk 1]
```diff
@@ -39,8 +39,13 @@ namespace tmx
...
     public:
 
-        explicit LayerGroup(const std::string& workDir, const Vector2u& tileCount);
+        LayerGroup(const std::string& workDir, const Vector2u& tileCount);
         ~LayerGroup() = default;
+		LayerGroup(const LayerGroup&) = delete;
+		const LayerGroup& operator = (const LayerGroup&) = delete;
+		LayerGroup(LayerGroup&&) = default;
+		LayerGroup& operator = (LayerGroup&&) = default;
+
```
Comment (Grant1219) [inline tmxlite/include/tmxlite/LayerGroup.hpp]: Interesting, why are these not required for the other `TileLayer`, `ObjectGroup`, or `ImageLayer` types? Also it looks like these lines are tabbed while all the others are indented with spaces.

Comment (fallahn) [reply tmxlite/include/tmxlite/LayerGroup.hpp]: Arg... they should be spaces! I'm going to blame VS for that :) I'm not sure of the specific reason msvc insists on these but it's down to the fact that the vector of layer pointers is uncopyable. It's the same with the `Map` class

**Comment by Grant1219** [comment]
Thanks for fixing the Windows build! I added a group layer to the test map and made sure to check that fetching properties and sublayers works. I realize not every edge case is tested, but increasing the test coverage for all the map elements might be a task for another PR.

Let me know if you want any further changes!

**Comment by fallahn** [comment]
Brilliant! All looks good, thanks for the contribution!
